streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -182,6 +182,8 @@ public String queryableStoreName() {
         final KTableProcessorSupplier<K, V, V> processorSupplier =
             new KTableFilter<>(this, predicate, filterNot, queryableStoreName);
 
+        processorSupplier.enableSendingOldValues(true);
```

Comment:
Why does the `filter` need to send old values? I though the `filter` needs to _receive_ old values, and thus we should call `this. enableSendingOldValues(true)` to enable sending old values on the filter's input?

Reply:
`KTableFilter` has a boolean flag internally that also needs to be set if it is to correctly handle old values, (existing code).

Reply:
SGMT.

IMHO, `sendOldValues` in `KTableFilter` should have only one meaning: do send old values downstream. If the filter result is materialized, we don't care if he upstream is sending old values or not. However, if the filter is stateless, as a side effect, we also need to tell upstream to send old values.

Reply:
> If the filter result is materialized, we don't care if he upstream is sending old values or not. However, if the filter is stateless, as a side effect, we also need to tell upstream to send old values.

That's not how the code was working. It always asked the parent to send old values:

```java
    @Override
    public void enableSendingOldValues() {
        parent.enableSendingOldValues();
        sendOldValues = true;
    }
```

Do we want to change this as part of this PR to only call `parent.enableSendingOldValues` is the filter is _not_ materialized?

Reply:
As discussed offline - change the behaviour of `enableSendingOldValues` such that a node that is already materialized will not ask upstream nodes to also send old values, is deemed out of scope of this PR.

@mjsax has requested a ticket to track this work.